lib/puppet/provider/windowsfeature/default.rb

```diff
@@ -0,0 +1,95 @@
+require 'json'
+Puppet::Type.type(:windowsfeature).provide(:default) do
+  # rubocop:disable Style/HashSyntax
+  commands :ps =>
+    if File.exist?("#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe")
+      "#{ENV['SYSTEMROOT']}\\sysnative\\WindowsPowershell\\v1.0\\powershell.exe"
+    elsif File.exist?("#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe")
+      "#{ENV['SYSTEMROOT']}\\system32\\WindowsPowershell\\v1.0\\powershell.exe"
+    else
+      'powershell.exe'
+    end
+
+  confine :kernel => :windows
+
+  def self.instances
+    features = JSON.parse(ps('Get-WindowsFeature | ConvertTo-JSON'))
+    features.collect do |feature|
+      name = feature['Name'].downcase
+      installed = feature['InstallState']
+      if installed == 1
+        currentstate = :present
+      elsif installed == 0
+        currentstate = :absent
+      end
+      new(:name => name, :ensure => currentstate)
+    end
+  end
+
+  def self.prefetch(resources)
+    features = instances
+    resources.keys.each do |name|
+      # rubocop:disable Lint/AssignmentInCondition
```

Comment:
This disables the cop for the rest of the file.  Probably best to put it at the end of the next line instead.